adagrams/game.py

```diff
@@ -1,11 +1,131 @@
+    
+    return letters #returns 10 randomly selected letters 
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    #loops through all the letters in the words and makes them lowercase strings in a new list
+    word_case_insensitive =  []
+    for i in word:
+        word_case_insensitive.append(i.lower())
+
+    #loops through all the letters in the letter bank and makes them lowercase strings in a new list
+    letter_bank_case_insensitive = []
+    for i in letter_bank:
+        letter_bank_case_insensitive.append(i.lower())
+
+    #counter for how many loops are completed
+    loop_count = 0
+
+    #for-loop that compares each item in both lists, and removes from letter bank if there's a match
+    for i in word_case_insensitive:
+        for j in letter_bank_case_insensitive:
+            if i == j:
+                loop_count += 1
+                letter_bank_case_insensitive.remove(j)
+
+    #if the loop count and length of a word are equal, it returns True and else False
+    if loop_count == len(word):
+        return True
+    else:
+        return False
+        
+            
+
 
 def score_word(word):
-    pass
+    #lists of scores grouped by number of points. Upper and lower are included to cover all bases (even though it might be not needed since the letters in the word are turned into lowercase)
+    score_1 = ['A', 'a', 'E', 'e', 'I', 'i', 'O', 'o', 'U', 'u', 'L', 'l','N', 'n', 'R', 'r', 'S','s', 'T','t']
+    score_2 = ['D', 'd', 'G', 'g']
+    score_3 = ['B', 'b', 'C', 'c', 'M', 'm','P', 'p']
+    score_4 = ['F', 'f', 'H', 'h', 'V', 'v', 'W', 'w','Y', 'y']
+    score_5 = ['K', 'k']
+    score_8 = ['J','j','X', 'x']
+    score_10 = ['Q', 'q','Z', 'z']
+
+    # Loops through all the letters in the word variable and makes them lowercase strings in a new list
+    word_ =  []
+    for i in word:
+        word_.append(i.lower())
+    
+    #variable is set and intalied to 0 to keep score
+    score = 0
+    #for each letter in 'word_' variable, if its in one of the lists it will add to the score accordingly 
+    for i in word_: 
+        if i in score_1:
+            score += 1
+        elif i in score_2:
+            score += 2
+        elif i in score_3:
+            score += 3  
+        elif i in score_4:
+            score += 4
+        elif i in score_5:
+            score += 5
+        elif i in score_8:
+            score += 8
+        elif i in score_10:
+            score += 10
+    
+    #if the length of the word is in between (and including) 7 or 10, it will add 8 points to the score total
+    if len(word_) <= 10 and len(word_) >= 7:
+        score += 8
+
+    #returns final score
+    return score
+        
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #word list is converted into tuple pairs. First is the word, second is the score called in from the score_word function for the word
```

Comment:
Good job with this one, it works well! Nice use of lambdas and `sorted` as well.